frontend/src/components/header/Header.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import {NavLink} from "react-router-dom";
+import './Header.css'
+import {HeaderProps} from "./props";
+
+const Header: React.FC<HeaderProps> = ({isLogged}: HeaderProps) => {
+    return (
+        <header className="header">
```

Comment:
You are trying hard to make it ugly. Material ui has lots of solutions for most of the common tasks https://mui.com/material-ui/react-app-bar/
this example is exactly what you need
https://mui.com/material-ui/react-app-bar/#app-bar-with-responsive-menu